File: computercraft/inspect_all.py
from .constants import Mock

turtle = Mock.Turtle()


def inspect_all():
    result = {}

    result["front"] = turtle.inspect()
    result["up"] = turtle.inspectUp()
    result["down"] = turtle.inspectDown()

    turtle.turnLeft()
    result["left"] = turtle.inspect()
    turtle.turnLeft()
    result["back"] = turtle.inspect()
    turtle.turnLeft()
    result["right"] = turtle.inspect()
    turtle.turnLeft()

    return result

File: computercraft/inspect_all.lua
function inspect_all()
    local result = {}

    result["front"] = turtle.inspect() or nil
    result["up"] = turtle.inspectUp() or nil
    result["down"] = turtle.inspectDown() or nil

    turtle.turnLeft()
    result["left"] = turtle.inspect() or nil
    turtle.turnLeft()
    result["back"] = turtle.inspect() or nil
    turtle.turnLeft()
    result["right"] = turtle.inspect() or nil
    turtle.turnLeft()

    return result
end


return inspect_all
